### PR TITLE
Add support for SCM integrations endpoint

### PR DESCRIPTION
**Pull request by EtienneM**
Related to Scalingo/dashboard-vue#271

### CODE REVIEW

File: src/SCMIntegrations/index.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  /**
+   * Fetch a specific SCM integration
+   * @see https://developers.scalingo.com/scm_integrations#show-a-specific-scm-integration
+   * @param {String} integrationID - ID of the integration
+   * @return {Promise<SCMIntegration,APIError>} Promise that when resolved returns a SCMIntegration
+   */
+  find(integrationID) {
+    return unpackData(
+      this._client.apiClient().get(`/scm_integrations/${integrationID}`),
```
Comment (johnsudaar) [inline src/SCMIntegrations/index.js]: Those request should be made to the auth service directly.

File: src/SCMIntegrations/index.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  }
+
+  /**
+   * Get all SCM integrations related to your account
+   * @see https://developers.scalingo.com/scm_integrations#list-all-the-scm-integrations-of-your-account
+   * @return {Promise<SCMIntegration[], APIError>} Promise that when resolved returns a SCMIntegration array.
+   */
+  all() {
+    return unpackData(
+      this._client.apiClient().get('/scm_integrations'),
```
Comment (johnsudaar) [inline src/SCMIntegrations/index.js]: Same here

File: src/SCMIntegrations/index.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+   * types are handled with this route.
+   * @see https://developers.scalingo.com/scm_integrations#createlink-an-scm-integration-with-your-account
+   * @param {SCMIntegrationCreateOpts} opts - SCM integration information
+   * @return {Promise<SCMIntegration, APIError>} Promise that when resolved returns the SCMIntegration created.
+   */
+  create(opts) {
+    return unpackData(
+      this._client
+        .apiClient()
+        .post('/scm_integrations', { scm_integration: opts }),
```
Comment (johnsudaar) [inline src/SCMIntegrations/index.js]: Same here

File: src/SCMIntegrations/index.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+  /**
+   * Delete/Unlink a SCM integration from your account.
+   * @see https://developers.scalingo.com/scm_integrations#createlink-an-scm-integration-with-your-account
+   * @param {String} integrationID - ID of the integration
+   * @return {Promise<null, APIError>} Promise that resolves when the SCM integration is deleted.
+   */
+  destroy(integrationID) {
+    return unpackData(
+      this._client.apiClient().delete(`/scm_integrations/${integrationID}`),
```
Comment (johnsudaar) [inline src/SCMIntegrations/index.js]: Same here

File: src/SCMIntegrations/index.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+  /**
+   * Import SSH keys from a SCM platform
+   * @see https://developers.scalingo.com/scm_integrations#createlink-an-scm-integration-with-your-account
+   * @param {String} integrationID - ID of the integration
+   * @return {Promise<SSHKey[], APIError>} Promise that when resolved returns the imported keys.
+   */
+  importSSHKeys(integrationID) {
+    return unpackData(
+      this._client
+        .apiClient()
```
Comment (johnsudaar) [inline src/SCMIntegrations/index.js]: same here

**Comment by EtienneM** [comment]
Silly me.... Sorry. Will do

**Comment by johnsudaar** [comment]
(plus rebase needed)

**Pull request merged**